pkgs/development/compilers/go/1.19.nix

```
@@ -141,13 +141,13 @@ stdenv.mkDerivation rec {
     # Contains the wrong perl shebang when cross compiling,
     # since it is not used for anything we can deleted as well.
     rm src/regexp/syntax/make_perl_groups.pl
-  '' + (if (stdenv.buildPlatform != stdenv.hostPlatform) then ''
+  '' + (if (stdenv.buildPlatform.system != stdenv.hostPlatform.system) then ''
```

If the intention is to allow this code here as it runs stuff built for host during the built, then this should be made clear properly (and you gain some cases not covered for free):

```suggestion
  '' + (if !(stdenv.buildPlatform.canExecute stdenv.hostPlatform) then ''
```

---

No, the problem is that when the go bootstrap tooling _thinks_ it's cross-compiling, it will put the result into a subdirectory /$target/.